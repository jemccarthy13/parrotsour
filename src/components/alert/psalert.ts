import {
  OptionsObject,
  SnackbarKey,
  closeSnackbar,
  enqueueSnackbar,
  SnackbarProvider,
} from "notistack"

export const snackActions = {
  success(msg: string, options: OptionsObject = {}): SnackbarKey {
    return enqueueSnackbar({ message: msg, ...options, variant: "success" })
  },
  warning(msg: string, options: OptionsObject = {}): SnackbarKey {
    return enqueueSnackbar({ message: msg, ...options, variant: "warning" })
  },
  info(msg: string, options: OptionsObject = {}): SnackbarKey {
    return enqueueSnackbar({ message: msg, ...options, variant: "info" })
  },
  error(msg: string, options: OptionsObject = {}): SnackbarKey {
    return enqueueSnackbar({ message: msg, ...options, variant: "error" })
  },
  closeSnackbar(key: SnackbarKey): void {
    closeSnackbar(key)
  },
  toast(msg: string, options: OptionsObject = {}): SnackbarKey {
    return enqueueSnackbar({ message: msg, ...options })
  },
}

export type { SnackbarKey }
export { SnackbarProvider }
