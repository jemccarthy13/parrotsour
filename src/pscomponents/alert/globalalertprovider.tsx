import React, { CSSProperties } from "react"

import { SnackbarKey, SnackbarProvider } from "notistack"

import snackActions, { SnackbarUtilsConfigurator } from "./psalert"

const transparentBtnStyle: CSSProperties = {
  height: "100%",
  left: 0,
  position: "absolute",
  top: 0,
  width: "100%",
  backgroundColor: "Transparent",
  backgroundRepeat: "no-repeat",
  outline: "none",
}

export default class GlobalAlertProvider extends SnackbarProvider {
  dismissNotification = (key: SnackbarKey): (() => void) => {
    return () => {
      snackActions.closeSnackbar(key)
    }
  }

  dismissAction = (key: SnackbarKey): React.ReactElement => {
    return (
      <button
        type="button"
        onClick={this.dismissNotification(key)}
        style={transparentBtnStyle}
      />
    )
  }

  render(): React.ReactElement {
    const { children, ...other } = this.props
    return (
      <SnackbarProvider {...other} action={this.dismissAction}>
        <SnackbarUtilsConfigurator />
        {children}
      </SnackbarProvider>
    )
  }
}
