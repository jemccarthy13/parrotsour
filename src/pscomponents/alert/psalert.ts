import {
  OptionsObject,
  SnackbarKey,
  useSnackbar,
  WithSnackbarProps,
} from "notistack"
import React from "react"

let snackbarRef: WithSnackbarProps
export const SnackbarUtilsConfigurator: React.FC = () => {
  snackbarRef = useSnackbar()
  return null
}

export default {
  success(msg: string, options: OptionsObject = {}): SnackbarKey {
    return this.toast(msg, { ...options, variant: "success" })
  },
  warning(msg: string, options: OptionsObject = {}): SnackbarKey {
    return this.toast(msg, { ...options, variant: "warning" })
  },
  info(msg: string, options: OptionsObject = {}): SnackbarKey {
    return this.toast(msg, { ...options, variant: "info" })
  },
  error(msg: string, options: OptionsObject = {}): SnackbarKey {
    return this.toast(msg, { ...options, variant: "error" })
  },
  closeSnackbar(key: SnackbarKey): void {
    snackbarRef.closeSnackbar(key)
  },
  toast(msg: string, options: OptionsObject = {}): SnackbarKey {
    return snackbarRef.enqueueSnackbar(msg, options)
  },
}
