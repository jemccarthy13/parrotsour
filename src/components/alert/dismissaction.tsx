import React, { useCallback } from "react"
import { Button } from "../../utils/muiadapter"
import { SnackbarKey, snackActions } from "./psalert"

type DismissActionProps = {
  snackKey: SnackbarKey
  confirmText?: string
  dismissText?: string
  confirmCallback: () => void
  cancelCallback: () => void
}
export function DismissAction({
  snackKey,
  confirmText = "Confirm",
  dismissText = "Dismiss",
  confirmCallback,
  cancelCallback,
}: Readonly<DismissActionProps>) {
  const handleConfirmClick = useCallback(() => {
    confirmCallback()
    snackActions.closeSnackbar(snackKey)
  }, [])

  const handleDismissClick = useCallback(() => {
    cancelCallback()
    snackActions.closeSnackbar(snackKey)
  }, [])

  return (
    <>
      <Button onClick={handleConfirmClick}>{confirmText}</Button>
      <Button onClick={handleDismissClick}>{dismissText}</Button>
    </>
  )
}
