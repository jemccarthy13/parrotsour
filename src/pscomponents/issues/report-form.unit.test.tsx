import React from "react"
import {
  RenderResult,
  act,
  fireEvent,
  render,
  waitFor,
} from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { snackActions } from "../alert/psalert"
import IssueReport from "./report-form"

jest.mock("../alert/psalert", () => ({
  snackActions: {
    success: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    toast: jest.fn(),
  },
}))

beforeAll(() => {
  console.warn(
    "1/15/2023- Surpressing MUI console.error for failure to wrap transition in act."
  )
  jest.spyOn(console, "error").mockImplementation()
})

describe("IssueReport_Component", () => {
  const answer = { pic: "2 GROUPS AZIMUTH 12", groups: [] }

  const openDialog = async (wrapper: RenderResult) => {
    await userEvent.click(wrapper.getByTestId(/iss-rpt-btn/i))
  }

  it("renders_correctly", () => {
    const wrapper = render(<IssueReport answer={answer} />)

    expect(wrapper).toBeDefined()
  })

  it("opens_dialog_on_button_click", async () => {
    const wrapper = render(<IssueReport answer={answer} />)

    await openDialog(wrapper)
    // open and content present
    expect(wrapper).toMatchSnapshot()
  })

  it("closes_dialog_on_cancel_click", async () => {
    const wrapper = render(<IssueReport answer={answer} />)

    act(() => {
      openDialog(wrapper)
    })

    await waitFor(() => {
      expect(wrapper.queryByText(/Cancel/)).not.toEqual(null)
    })

    userEvent.click(wrapper.getByText(/Cancel/))
    expect(wrapper).toMatchSnapshot()
  })

  it("closes_on_clickaway", async () => {
    const wrapper = render(
      <div id="root">
        <IssueReport answer={answer} />
      </div>
    )

    act(() => {
      openDialog(wrapper)
    })

    await waitFor(() => {
      expect(wrapper.queryByText(/Cancel/)).not.toEqual(null)
    })
    const dialogBtnRoot = wrapper.getByText(/Cancel/)

    fireEvent.keyDown(dialogBtnRoot, {
      key: "Escape",
      code: "Escape",
      keyCode: 27,
      charCode: 27,
    })

    await waitFor(() => {
      expect(wrapper.queryAllByText(/Cancel/).length).toEqual(0)
    })
  })

  it("handles_cancel", async () => {
    const wrapper = render(<IssueReport answer={answer} />)

    act(() => {
      openDialog(wrapper)
    })

    await waitFor(() => {
      expect(wrapper.getByText(/Cancel/)).not.toEqual(null)
    })

    const cancelBtn = wrapper.getByText(/Cancel/)

    act(() => {
      userEvent.click(cancelBtn)
    })

    await waitFor(() => {
      expect(wrapper.queryAllByText(/Cancel/).length).toEqual(0)
    })
  })

  it("handles_email_field_changes", async () => {
    const wrapper = render(<IssueReport answer={answer} />)

    act(() => {
      openDialog(wrapper)
    })

    await waitFor(() => {
      expect(wrapper.getByText(/Cancel/)).not.toEqual(null)
    })

    const emailBox = wrapper.getByRole(/textbox/, { name: "Email" })

    emailBox.focus()

    fireEvent.change(emailBox, { target: { value: "abcdefg" } })

    await waitFor(() => {
      expect((emailBox as HTMLTextAreaElement).value).toEqual("abcdefg")
    })
  })

  it("handles_descr_field_changes", async () => {
    const wrapper = render(<IssueReport answer={answer} />)

    act(() => {
      openDialog(wrapper)
    })

    await waitFor(() => {
      expect(wrapper.getByText(/Cancel/)).not.toEqual(null)
    })

    const emailBox = wrapper.getByRole(/textbox/, { name: "Issue Description" })

    emailBox.focus()

    fireEvent.change(emailBox, { target: { value: "issue text here" } })

    await waitFor(() => {
      expect((emailBox as HTMLTextAreaElement).value).toEqual("issue text here")
    })
  })

  it("handles_submit", async () => {
    const wrapper = render(<IssueReport answer={answer} />)

    const snackSpy = jest
      .spyOn(snackActions, "error")
      .mockImplementation(jest.fn())

    act(() => {
      openDialog(wrapper)
    })

    await waitFor(() => {
      expect(wrapper.getByText(/Cancel/)).not.toEqual(null)
    })

    act(() => {
      userEvent.click(wrapper.getByTestId(/iss-oth-selector/))
    })

    await waitFor(() => {
      expect(
        (wrapper.getByTestId(/iss-oth-selector/) as HTMLInputElement).checked
      ).toEqual(true)
    })
    const email = wrapper.getByRole(/textbox/, { name: "Email" })

    fireEvent.change(email, { target: { value: "abc@def.com" } })

    const issTxt = wrapper.getByRole(/textbox/, { name: "Issue Description" })

    fireEvent.change(issTxt, { target: { value: "My issue description" } })

    await waitFor(() => {
      expect((issTxt as HTMLInputElement).value).toEqual("My issue description")
    })

    const submitBtn = wrapper.getByRole(/button/, { name: "Submit" })

    userEvent.click(submitBtn)

    await waitFor(() => {
      expect(snackSpy).toHaveBeenCalled()
    })
  })
})
