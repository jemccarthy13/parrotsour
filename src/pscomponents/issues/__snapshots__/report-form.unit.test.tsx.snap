// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueReport_Component closes_dialog_on_cancel_click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        data-testid="iss-rpt-form"
        style="width: 25%;"
      >
        <button
          data-testid="iss-rpt-btn"
          id="showFormBtn"
          style="margin-left: 5%; top: 5px;"
          type="button"
        >
          Report Issue
        </button>
      </div>
    </div>
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r4:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            See a list of 
            <a
              href="/#/changelog.html"
            >
              known issues
            </a>
            .
          </div>
          <form>
            <div
              class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
            >
              <div
                class="pscontainer"
              >
                <h2>
                  <u>
                    Problem:
                  </u>
                </h2>
                <ul>
                  <li>
                    <input
                      checked=""
                      data-testid="iss-pic-selector"
                      id="picprob"
                      name="issue"
                      style="color: grey;"
                      type="radio"
                      value="picprob"
                    />
                    <label
                      for="picprob"
                      id="piclbl"
                      style="color: grey;"
                    >
                      The answer to this picture is incorrect
                    </label>
                    <div
                      class="check-no-hover"
                    />
                  </li>
                  <li>
                    <input
                      data-testid="iss-feature-selector"
                      id="feature"
                      name="issue"
                      style="color: grey;"
                      type="radio"
                      value="feature"
                    />
                    <label
                      for="feature"
                      style="color: grey;"
                    >
                      Wouldn‘t it be nice if...
                    </label>
                    <div
                      class="check-no-hover"
                    />
                  </li>
                  <li>
                    <input
                      data-testid="iss-oth-selector"
                      id="othprob"
                      name="issue"
                      type="radio"
                      value="othprob"
                    />
                    <label
                      for="othprob"
                      style="color: grey;"
                    >
                      Something else
                    </label>
                    <div
                      class="check-no-hover"
                    />
                  </li>
                </ul>
              </div>
            </div>
            <div
              style="width: 75%; margin: auto; height: 100%;"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Email
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="email"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Email
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="issue"
                  id="issue-label"
                >
                  Issue Description
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    id="issue"
                    name="issuetxt"
                    required=""
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Issue Description
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                id="submitIssueReport"
                tabindex="0"
                type="button"
              >
                Submit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                id="cancelIssueReport"
                tabindex="0"
                type="button"
              >
                Cancel
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      data-testid="iss-rpt-form"
      style="width: 25%;"
    >
      <button
        data-testid="iss-rpt-btn"
        id="showFormBtn"
        style="margin-left: 5%; top: 5px;"
        type="button"
      >
        Report Issue
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`IssueReport_Component opens_dialog_on_button_click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        data-testid="iss-rpt-form"
        style="width: 25%;"
      >
        <button
          data-testid="iss-rpt-btn"
          id="showFormBtn"
          style="margin-left: 5%; top: 5px;"
          type="button"
        >
          Report Issue
        </button>
      </div>
    </div>
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r1:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            See a list of 
            <a
              href="/#/changelog.html"
            >
              known issues
            </a>
            .
          </div>
          <form>
            <div
              class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
            >
              <div
                class="pscontainer"
              >
                <h2>
                  <u>
                    Problem:
                  </u>
                </h2>
                <ul>
                  <li>
                    <input
                      checked=""
                      data-testid="iss-pic-selector"
                      id="picprob"
                      name="issue"
                      style="color: grey;"
                      type="radio"
                      value="picprob"
                    />
                    <label
                      for="picprob"
                      id="piclbl"
                      style="color: grey;"
                    >
                      The answer to this picture is incorrect
                    </label>
                    <div
                      class="check-no-hover"
                    />
                  </li>
                  <li>
                    <input
                      data-testid="iss-feature-selector"
                      id="feature"
                      name="issue"
                      style="color: grey;"
                      type="radio"
                      value="feature"
                    />
                    <label
                      for="feature"
                      style="color: grey;"
                    >
                      Wouldn‘t it be nice if...
                    </label>
                    <div
                      class="check-no-hover"
                    />
                  </li>
                  <li>
                    <input
                      data-testid="iss-oth-selector"
                      id="othprob"
                      name="issue"
                      type="radio"
                      value="othprob"
                    />
                    <label
                      for="othprob"
                      style="color: grey;"
                    >
                      Something else
                    </label>
                    <div
                      class="check-no-hover"
                    />
                  </li>
                </ul>
              </div>
            </div>
            <div
              style="width: 75%; margin: auto; height: 100%;"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Email
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="email"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Email
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="issue"
                  id="issue-label"
                >
                  Issue Description
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    id="issue"
                    name="issuetxt"
                    required=""
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Issue Description
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                id="submitIssueReport"
                tabindex="0"
                type="button"
              >
                Submit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                id="cancelIssueReport"
                tabindex="0"
                type="button"
              >
                Cancel
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      data-testid="iss-rpt-form"
      style="width: 25%;"
    >
      <button
        data-testid="iss-rpt-btn"
        id="showFormBtn"
        style="margin-left: 5%; top: 5px;"
        type="button"
      >
        Report Issue
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
